comptests/prov/test_prov_k8s_cortx_cold_upgrade.py

```diff
@@ -0,0 +1,210 @@
+            cls.host_list.append(node_obj)
+            if CMN_CFG["nodes"][node]["node_type"].lower() == "master":
+                cls.master_node_obj = node_obj
+                cls.master_node_list.append(node_obj)
+            else:
+                cls.worker_node_list.append(node_obj)
+        LOGGER.info("Done: Setup operations finished.")
+
+    def perform_upgrade(self, exc: bool = True, output=None):
+        """Function calls upgrade and put return value in queue object."""
+        LOGGER.info("Calling upgrade.")
+        resp = self.deploy_lc_obj.upgrade_software(self.master_node_obj,
+                                                   self.prov_deploy_cfg["git_remote_path"],
+                                                   exc=exc)
+        output.put(resp)
+
+    @pytest.mark.run(order=1)
+    @pytest.mark.lc
+    @pytest.mark.comp_prov
+    @pytest.mark.tags("TEST-37357")
+    def test_37357(self):
+        """
+        Verify CORTX Software Cold upgrade.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Step 1: Get installed version.")
+        resp = HAK8s.get_config_value(self.master_node_obj)
+        assert_utils.assert_true(resp[0], resp[1])
+        installed_version = resp[1]['cortx']['common']['release']['version']
+        LOGGER.info("Current version: %s", installed_version)
+        LOGGER.info("Step 1: Done.")
+
+        LOGGER.info("Step 2: Check if installing version is higher than installed version.")
+        # TODO : Better way to compare two versions.
+        installing_version = self.cortx_all_image.split(":")[1].split("-")
+        installing_version = installing_version[0] + "-" + installing_version[1]
+        LOGGER.info("Installing CORTX image verson: %s", installing_version)
+        assert installing_version > installed_version, \
+            "Installed version is higher than installing version."
+        LOGGER.info("Step 2: Done.")
+
+        LOGGER.info("Step 3: Check cluster health.")
+        resp = self.deploy_lc_obj.check_s3_status(self.master_node_obj)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 3: Done.")
+
+        LOGGER.info("Step 4: Change image version in solution.yaml.")
+        remote_sol_path = self.prov_deploy_cfg["git_remote_path"] + "solution.yaml"
+        solution_path = self.master_node_obj.copy_file_to_local(remote_path=remote_sol_path,
+                                                                local_path=self.local_sol_path)
+        assert_utils.assert_true(solution_path[0], solution_path[1])
+        image_dict = {"all_image": self.cortx_all_image, "rgw_image": self.cortx_rgw_image}
+        local_path = self.deploy_lc_obj.update_sol_with_image(self.local_sol_path, image_dict)
+        assert_utils.assert_true(local_path[0], local_path[1])
+        for node_obj in self.host_list:
+            resp = self.deploy_lc_obj.copy_sol_file(node_obj, local_sol_path=local_path[1],
```

Comment:
Is the solution file required to be copied on all the worker nodes ?

Reply:
for solution.yaml we ideally copy to all worker nodes also.